### PR TITLE
Remove multisig's Calls

### PR DESCRIPTION
**Pull request by gavofyork**
Just removes the `Calls` item and its associated functionality, needed prior to 2D-weights integration.

## TODO
- [x] Fix benchmarks
- [x] Fix tests
- [x] Consider migration to refund and clear `Calls`

### CODE REVIEW

**Comment by ruseinov** [comment]
bot rebase

**Comment by ruseinov** [comment]
bot rebase

File: frame/multisig/src/lib.rs [hunk 1]
```diff
@@ -639,15 +603,6 @@ impl<T: Config> Pallet<T> {
 			// Just start the operation by recording it in storage.
 			let deposit = T::DepositBase::get() + T::DepositFactor::get() * threshold.into();
 
-			// Store the call if desired.
-			let stored = if let Some(data) = maybe_call.filter(|_| store) {
-				Self::store_call_and_reserve(who.clone(), &call_hash, data, deposit)?;
-				true
-			} else {
-				T::Currency::reserve(&who, deposit)?;
```
Comment (ruseinov) [inline frame/multisig/src/lib.rs:647]: @kianenigma is the right? I'm not sure why the reserve has been removed as it has nothing to do with the storage removal.

Comment (ruseinov) [reply frame/multisig/src/lib.rs:647]: @gavofyork could you shine some light on this please? I'm in the middle of redoing tests and benches and this is crucial to removing some of the redundant tests. If this was a mistake - I can re-introduce the reserve.

Comment (ruseinov) [reply frame/multisig/src/lib.rs:647]: I am assuming this is a bug, I will reintroduce the deposit in another PR along with fixed tests

**Comment by kianenigma** [comment]
being worked on here https://github.com/paritytech/substrate/pull/12072

**Comment by ruseinov** [comment]
The only thing left is the migration.

**Comment by ruseinov** [comment]
bot rebase

**Comment by ruseinov** [comment]
bot rebase

**Comment by ruseinov** [comment]
@kianenigma @gavofyork @ggwpez let's give it a final look-through and merge into it's parent branch, then we can merge this https://github.com/paritytech/polkadot/pull/6075 into the companion.

File: frame/multisig/src/benchmarking.rs [hunk 2]
```diff
@@ -237,45 +188,6 @@ benchmarks! {
 		assert_eq!(multisig.approvals.len(), 2);
 	}
 
-	approve_as_multi_complete {
```
Comment (ggwpez) [inline frame/multisig/src/benchmarking.rs:240]: @ruseinov can you help me to understand why this is not needed anymore?

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis/1026/2

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-digest-11-jan-2023/1654/1

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-34/1432/8

**Pull request merged**